test/integration/toolchains/cmake/test_cmaketoolchain.py

```diff
@@ -1278,6 +1278,32 @@ def layout(self):
     assert "conan-windows-pkg-0.1-debug" in presets
 
 
+def test_build_folder_vars_constants_user():
+    c = TestClient()
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.cmake import cmake_layout
+
+        class Conan(ConanFile):
+            name = "dep"
+            version = "0.1"
+            settings = "os", "build_type"
+            generators = "CMakeToolchain"
+
+            def layout(self):
+                cmake_layout(self)
+        """)
+    c.save({"conanfile.py": conanfile})
+    conf = "tools.cmake.cmake_layout:build_folder_vars='[\"user.myvalue\"]'"
```

Comment:
I would be ok with this but I was thinking, why don't we use a user conf to set the value for this?
Like:

```
conan install . -c=tools.cmake.cmake_layout:build_folder_vars='["user.whatever:myvar"]' -s build_type=Debug -c user.whatever:myvar=pepito
```

For me this feels more aligned with the rest of things that we can set via `build_folder_vars` that are variable names that are later evaluated to get the value, not the values directly.

Reply:
That is true, I thought about it, but I checked most of the existing configurations, it didn't seem any of them made much sense, so it should be only ``user.``, not ``tools.`` or ``core.``, and then it was just extra verbosity to define a constant. I actually thought to call it ``constant.myvalue`` to differentiate from ``conf`` ``user.`` values.

Reply:
I think calling it `constant.` is a great idea!

Reply:
Some ideas...

```
param.myvalue
const.myvalue
prop.myvalue
var.myvalue
var.myvalue
value.myvalue
value.myvalue
```

Maybe we can use `:` instead of `.` ? like `const:myvalue`

Also what about using a symbol preceding the value?

```
@myvalue
::myvalue
#myvalue
```

Reply:
I'd vote for ``const.myvalue``.

The syntax would be more homogenous: 
```
build_folder_vars = [settings.os, options.shared, self.name, const.myvalue]
```

Reply:
I think `const.myvalue` is good enough 👍

Reply:
👍 for const

Reply:
Done